src/platforms/web/runtime/transition-util.js

```diff
@@ -68,6 +68,10 @@ export function nextFrame (fn: Function) {
   })
 }
 
+export function hasTransitionClass (el: any, cls: string) {
```

Comment:
Put `hasTransitionClass` check only inside `nextFrame` is trying to avoid unnecessary checking. It fair to put it inside addTransitionClass.